src/Raven.Server/Config/JsonConfigFileModifier.cs

```diff
@@ -0,0 +1,185 @@
+            var dicConfig = new Dictionary<string, object>();
+            foreach (var (key, value) in orderConfig)
+            {
+                if (value == null)
+                    continue;
+                
+                var directKey = key.Replace(':', '.');
+                if (dicConfig.TryAdd(directKey, value) == false)
+                    continue;
+
+                jsonConfig[directKey] = value;
+            }
+        }
+        catch (FileNotFoundException)
+        {
+        }
+    }
+
+    private static FileStream OpenFile(string path, FileMode fileMode, FileAccess fileAccess)
+    {
+        try
+        {
+            return SafeFileStream.Create(path, fileMode, fileAccess);
+        }
+        catch (Exception e) when (e is UnauthorizedAccessException or SecurityException)
+        {
+            throw new UnsuccessfulFileAccessException(e, path, fileAccess);
+        }
+    }
+
+    private async Task PersistConfigurationAsync(BlittableJsonReaderObject json)
+    {
+        var tempFile = _path + ".tmp";
+
+        try
+        {
+            await using (var file = OpenFile(tempFile, FileMode.Create, FileAccess.ReadWrite))
+            await using (var streamWriter = new StreamWriter(file))
+            await using (var writer = new JsonTextWriter(streamWriter))
+            await using (var reader = new BlittableJsonReader())
+            {
+                writer.Formatting = Formatting.Indented;
+                reader.Initialize(json);
+
+                await writer.WriteTokenAsync(reader);
+
+                await writer.FlushAsync();
+                await streamWriter.FlushAsync();
+                file.Flush(true);
+            }
+
+            Validate(tempFile);
+            SwitchTempWithOriginalAndCreateBackup(tempFile);
+        }
+        finally
+        {
+            try
+            {
+                File.Delete(tempFile);
+            }
+            catch
+            {
+                // ignored
+            }
+        }
+    }
+
+    private void SwitchTempWithOriginalAndCreateBackup(string tempPath)
+    {
+        try
+        {
+            if (File.Exists(_path))
+            {
+                File.Replace(tempPath, _path, _path + ".bak");
+            }
+            else
+            {
+                File.Move(tempPath, _path);
+            }
+            
+            if (PlatformDetails.RunningOnPosix)
+                Syscall.FsyncDirectoryFor(_path);
+        }
+        catch (Exception e) when (e is UnauthorizedAccessException or SecurityException)
+        {
+            throw new UnsuccessfulFileAccessException(e, _path, FileAccess.Write);
+        }
+    }
+
+    protected bool IsOriginalValue<T>(string key, T value)
+    {
+        if (_originJson.TryGetMember(key, out object result) == false)
+            return false;
+        if(BlittableJsonReaderObject.TryConvertType<T>(result, out var tValue) == false)
+        {
+            if (Logger.IsOperationsEnabled)
```

Comment:
is operations here, and Info one line below

Reply:
OMG, sorry.
I am also adding a `Debug.Assert` here